administrator/modules/mod_menu/menu.php

```diff
@@ -380,7 +380,7 @@ protected function preprocess($items)
 			}
 
 			// Exclude if link is invalid
-			if (!in_array($item->type, array('separator', 'heading', 'container')) && trim($item->link) === '')
+			if (!in_array($item->type, array('separator', 'heading', 'container')) && trim((string) $item->link) === '')
```

Comment:
```suggestion
			if (!in_array($item->type, array('separator', 'heading', 'container')) && empty($item->link))
```

Isnt an empty check better here vs trim and `=== ''`

Reply:
Empty will not be true when the string is just a bunch of spaces so we need the trim, or am I wrong?

Reply:
With empty also [$item->link = null](https://3v4l.org/9Jdhb) will fail the test here with [trim it would have passed this test](https://3v4l.org/p8fg0). But to me $item->link = null should be catched here too right?

Reply:
Personal taste: I don't like using `empty()` as it's basically same as `(! (bool) $value)` just that if `$value` doesn't exist, it considers it "empty" too. 
https://www.php.net/manual/en/function.empty.php

Additionally, string '0' is considered as empty too.
Additionally, iirc typecastings have slightly changed over the lifecycle of PHP and might change in the future.

So as personal taste, I don't find `empty()` to be a type-clean function. Thus I'm not willing to take the risk of using empty() as suggested.

My change was a BC safe change because it does exactly what PHP does anyway, without changing anything in the behavior. If casting to (string) is not acceptable, I'm ok to merge a different way, with `is_null($item->type)` which would then give valid warnings for non-null non-string $item->link.

Updated PR with the slightly cleaner and equivalent fix.

Reply:
Side note: After reading again, my reply was way too factual and might read as being hard factual. Please accept my appologizes. I'm non-native English. The comment doesn't convey properly my gratitude. Fixing it here:

Thank you @zero-24 and @richard67 for your feedbacks and aim to have Joomla as clean as possible! :+1:

Reply:
No problem and thanks for your PRs for PHP8.1